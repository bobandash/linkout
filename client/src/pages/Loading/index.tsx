import { Audio } from 'react-loader-spinner';

const Loading = () => {
  return (
    <div className="bg-color_3 min-h-screen w-screen">
      <div className="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2">
        <Audio height="80" width="80" color="white" ariaLabel="loading" />
      </div>
    </div>
  );
};

export default Loading;
