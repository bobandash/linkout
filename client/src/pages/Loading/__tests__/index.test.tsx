import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Loading from '../index';

describe('Loading Screen', () => {
  it('should render loading screen', () => {
    const container = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
