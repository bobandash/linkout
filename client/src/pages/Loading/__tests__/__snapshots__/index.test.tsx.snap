// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading Screen > should render loading screen 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative min-h-screen w-full bg-color_3"
      >
        <div
          class="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            data-testid="audio-loading"
            role="status"
            style="display: flex;"
          >
            <svg
              data-testid="audio-svg"
              fill="white"
              height="80"
              viewBox="0 0 55 80"
              width="80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                transform="matrix(1 0 0 -1 0 80)"
              >
                <rect
                  height="20"
                  rx="3"
                  width="10"
                >
                  <animate
                    attributeName="height"
                    begin="0s"
                    calcMode="linear"
                    dur="4.3s"
                    repeatCount="indefinite"
                    values="20;45;57;80;64;32;66;45;64;23;66;13;64;56;34;34;2;23;76;79;20"
                  />
                </rect>
                <rect
                  height="80"
                  rx="3"
                  width="10"
                  x="15"
                >
                  <animate
                    attributeName="height"
                    begin="0s"
                    calcMode="linear"
                    dur="2s"
                    repeatCount="indefinite"
                    values="80;55;33;5;75;23;73;33;12;14;60;80"
                  />
                </rect>
                <rect
                  height="50"
                  rx="3"
                  width="10"
                  x="30"
                >
                  <animate
                    attributeName="height"
                    begin="0s"
                    calcMode="linear"
                    dur="1.4s"
                    repeatCount="indefinite"
                    values="50;34;78;23;56;23;34;76;80;54;21;50"
                  />
                </rect>
                <rect
                  height="30"
                  rx="3"
                  width="10"
                  x="45"
                >
                  <animate
                    attributeName="height"
                    begin="0s"
                    calcMode="linear"
                    dur="2s"
                    repeatCount="indefinite"
                    values="30;45;13;80;56;72;45;76;34;23;67;30"
                  />
                </rect>
              </g>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative min-h-screen w-full bg-color_3"
    >
      <div
        class="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2"
      >
        <div
          aria-busy="true"
          aria-label="loading"
          data-testid="audio-loading"
          role="status"
          style="display: flex;"
        >
          <svg
            data-testid="audio-svg"
            fill="white"
            height="80"
            viewBox="0 0 55 80"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="matrix(1 0 0 -1 0 80)"
            >
              <rect
                height="20"
                rx="3"
                width="10"
              >
                <animate
                  attributeName="height"
                  begin="0s"
                  calcMode="linear"
                  dur="4.3s"
                  repeatCount="indefinite"
                  values="20;45;57;80;64;32;66;45;64;23;66;13;64;56;34;34;2;23;76;79;20"
                />
              </rect>
              <rect
                height="80"
                rx="3"
                width="10"
                x="15"
              >
                <animate
                  attributeName="height"
                  begin="0s"
                  calcMode="linear"
                  dur="2s"
                  repeatCount="indefinite"
                  values="80;55;33;5;75;23;73;33;12;14;60;80"
                />
              </rect>
              <rect
                height="50"
                rx="3"
                width="10"
                x="30"
              >
                <animate
                  attributeName="height"
                  begin="0s"
                  calcMode="linear"
                  dur="1.4s"
                  repeatCount="indefinite"
                  values="50;34;78;23;56;23;34;76;80;54;21;50"
                />
              </rect>
              <rect
                height="30"
                rx="3"
                width="10"
                x="45"
              >
                <animate
                  attributeName="height"
                  begin="0s"
                  calcMode="linear"
                  dur="2s"
                  repeatCount="indefinite"
                  values="30;45;13;80;56;72;45;76;34;23;67;30"
                />
              </rect>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
